### PR TITLE
Use Environment Variables for API Base URL

### PR DESCRIPTION
**Pull request by tanya-newStart**
This PR replaces the hardcoded localhost API URL with environment variables to support different environments.

Added .env.local and .env.production:
.env.local: NEXT_PUBLIC_API_BASE_URL=http://localhost:3001
.env.production: NEXT_PUBLIC_API_BASE_URL=https://leafnotes.onrender.com

Updated code to use process.env.NEXT_PUBLIC_API_BASE_URL for API requests.

### CODE REVIEW

File: app/app/utils/makeRequest.js [hunk 1]
```diff
@@ -1,6 +1,11 @@
-export const makeRequest = async (url, userData = {}, method = "POST") => {
+export const makeRequest = async (endpoint, userData = {}, method = "POST") => {
+  const baseUrl = process.env.NEXT_PUBLIC_API_BASE_URL;
+  console.log(baseUrl);
```
Comment (magdazelena) [inline app/app/utils/makeRequest.js]: is this here on purpose? 👀

**Review by magdazelena — APPROVED** [review]
really nice! just a small thingy :)

File: app/app/utils/makeRequest.js (hunk 1 above)
Comment (tanya-newStart) [reply app/app/utils/makeRequest.js]: thank you! :)

File: app/app/components/AddBookToBookshelf.jsx [hunk 2]
```diff
@@ -1,15 +1,58 @@
...
+
+const AddBookToBookshelf = ({ category, onBookAdded, bookShelf }) => {
   const [query, setQuery] = useState("");
   const [searchResults, setSearchResults] = useState([]);
   const [loading, setLoading] = useState(false);
   const [error, setError] = useState(null);
+  const [successMessage, setSuccessMessage] = useState(null);
+
+  // Debounced search function
+  const debouncedFetchSearchResults = debounce(
```
Comment (magdazelena) [inline app/app/components/AddBookToBookshelf.jsx:25]: 👏

**Review by magdazelena — APPROVED** [review]
nice work :)

**Pull request merged**